semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,597 @@
+import sys
+from collections import namedtuple
+import os.path
+from datetime import datetime as dt
+import numpy as np
+from scipy.interpolate import CloughTocher2DInterpolator
+
+from ecl.grid import EclGrid
+from ecl.gravimetry import EclSubsidence
+from ecl.eclfile import Ecl3DKW, EclFile
+
+from ecl.util.geometry import Surface
+
+from .ots_vel_surface import OTSVelSurface
+from .ots_res_surface import OTSResSurface
+from .ots_config import build_schema
+import configsuite
+import yaml
+
+
+def extract_ots_context(configuration):
+    if configuration.eclbase is not None:
+        rstfile_path = "{}.UNRST".format(configuration.eclbase)
+        rstfile = EclFile(rstfile_path)
+        dates = [d.date() for d in rstfile.dates]
+        return dates
+    return []
+
+
+def ots_load_params(input_file):
+    config = None
+    with open(input_file, "r") as fin:
+        config = yaml.safe_load(fin)
+    config = configsuite.ConfigSuite(
+        config,
+        build_schema(),
+        deduce_required=True,
+        extract_validation_context=extract_ots_context,
+    )
+    assert config.valid
+
+    input_data = config.snapshot
+    return input_data, input_data.vintages
+
+
+def write_surface(vintage_pairs, ts, output_dir, type_str):
+    for iv, vp in enumerate(vintage_pairs):
+        d0 = vp[0].strftime("%Y_%m_%d")
+        d1 = vp[1].strftime("%Y_%m_%d")
+        ts[iv].write(output_dir + type_str + "/ots" + "_" + d0 + "_" + d1 + ".irap")
+
+
+def ots_run(parameter_file, verbose=False):
+
+    parms, vintage_pairs = ots_load_params(parameter_file)
+
+    ots = OTS(
+        eclbase=parms.eclbase,
+        mapaxes=parms.mapaxes,
+        seabed=parms.seabed,
+        youngs=parms.youngs,
+        poisson=parms.poisson,
+        rfactor=parms.rfactor,
+        convention=parms.convention,
+        above=parms.above,
+        velocity_model=parms.velocity_model,
+        verbose=verbose,
+    )
+
+    tshift_ts = ots.geertsma_ts(vintage_pairs.ts)
+    tshift_ts_simple = ots.geertsma_ts_simple(vintage_pairs.ts_simple)
+    tshift_dpv = ots.dpv(vintage_pairs.dpv)
+    tshift_ts_rporv = ots.geertsma_ts_rporv(vintage_pairs.ts_rporv)
+
+    surface_horizon = ots.get_horizon()
+
+    write_surface(vintage_pairs.ts, tshift_ts, parms.output_dir, "_ts")
+    write_surface(
+        vintage_pairs.ts_simple, tshift_ts_simple, parms.output_dir, "_ts_simple"
+    )
+    write_surface(vintage_pairs.dpv, tshift_dpv, parms.output_dir, "_dpv")
+    write_surface(
+        vintage_pairs.ts_rporv, tshift_ts_rporv, parms.output_dir, "_ts_rporv"
+    )
+
+    if parms.horizon is not None:
+        surface_horizon.write(parms.horizon)
+
+    num_pairs = (
+        len(vintage_pairs.ts)
+        + len(vintage_pairs.ts_simple)
+        + len(vintage_pairs.dpv)
+        + len(vintage_pairs.ts_rporv)
+    )
+    line = "{}, {}, {}" + ", {}" * num_pairs + "\n"
+
+    if parms.ascii is not None:
+        with open(parms.ascii, "w") as f:
```

Comment:
Split this writing out into a separate function as well, as you've done with the surface writing?

Also, I can't help but think that there must be a better way of writing this.. :P 

I'll look at the rest and come back here afterwards for suggestion

Reply:
Agree, there could be a better way to writing this to a file, but I think this would require also a better data structure in the first place. So I left it as is currently, let's get back to it.